src/PkgTasks.h

```diff
@@ -121,7 +121,7 @@ class PkgTask
      * Return 'true' if this action was requested by the user and not
      * automatically to satisfy dependencies.
      **/
-    bool byUser() const { return _requester | PkgReqUser; }
+    bool byUser() const { return ( _requester == PkgReqUser ); }
```

Comment:
The binary `|` (_or_) was indeed wrong; this should be a binary `&` (_and_). But that's not the same as `==`; multiple bits may be set at the same time, and `==` is only `true` if the exact same bits are set, unlike `&` that returns a `true` value if that bit is set, regardless of other bits.

Reply:
Aww, you are right. I missed the case `PkgReqAll`. I adjusted the code accordingly.